tooling/helmtest/main_test.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2025 Microsoft Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	repoRootDir = "../../"
+)
+
+type testCases struct {
+	Cases []testCase `yaml:"cases"`
+}
+
+type testCase struct {
+	Name         string `yaml:"name"`
+	Namespace    string `yaml:"namespace"`
+	SetFile      string `yaml:"setFile"`
+	HelmChartDir string `yaml:"helmChartDir"`
+}
+
+func getTestCases() (*testCases, error) {
+	helmtestsPath := filepath.Join(repoRootDir, ".helmtests.yaml")
+	testConfigBytes, err := os.ReadFile(helmtestsPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read .helmtests.yaml: %w", err)
+	}
+	var testConfig *testCases
+	err = yaml.Unmarshal(testConfigBytes, &testConfig)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshal .helmtests.yaml: %w", err)
+	}
+	return testConfig, nil
+}
+
+func runHelmTemplate(helmDeploy, helmChartDir, namespace string, setValues []string) (string, error) {
+	args := []string{
+		"helm",
+		"template",
```

Comment:
Import & use their library :)

Reply:
I'd rather not. The template command is not exported, as i.e. the upgrade. I'd have to re-implement the templating here.

Reply:
I will revisit this, since using the SDK will also make it easier to create tests and reuse the config/values files!